### ISSUE TITLE
Annotate API with javax.annotation.Nullable/Nonnull for Kotlin Interop 

### ISSUE THREAD

**Issue by JLLeitschuh** [google/guice#1159]
Hi, maintainer of [Kotlin-Guiced](https://github.com/JLLeitschuh/kotlin-guiced) here.

It would be really nice if Guice packages/classes were annotated with the `javax.annotation.*` annotations.

These are the various ways that the Kotlin language handles Null annotations. If the whole API is null safe then you can simply slap an annotation on the `package-info.java` file and that's it.
https://kotlinlang.org/docs/reference/java-interop.html#nullability-annotations

Without these the Kotlin compiler gives a lot of warnings about the library not specifying forcing developers to explicitly null check in order to remove the warning; often in places where checking for null isn't required because the library won't return a null value.

Is this something that the team would consider doing? If not, would you accept a PR adding this?

**Comment by lukesandberg** [comment]
Wasn’t the JSR for these annotations abandoned?  Is it a good idea to standardize on them?

**Comment by kashike** [comment]
Related: https://github.com/google/guava/issues/2960

**Comment by drothmaler** [comment]
Yes, you are right, javax.annotation.* / JSR305 are probably not a good choice any more (especially because of the Module System incompatibility). But Kotlin can infer nullability from a lot of other types of Nullable/NonNull annotation (as described in the link to the Kotlin documentation, provided in the initial post). Personally I tend to use the annotations from the CheckerFramework (`org.checkerframework.checker.nullness.qual.Nullable` / `org.checkerframework.checker.nullness.qual.NonNull`) because they feel most neutral to me.
